### PR TITLE
[hap2] Create new nagios plugin that use livestatus API

### PR DESCRIPTION
**Pull request by Mnakagawa**
livestatus API:
https://mathias-kettner.de/checkmk_livestatus.html

CentOS(from epel repository)
```
# yum install check-mk-livestatus
```

Ubuntu
```
# apt-get install check-mk-livestatus
```

Python library:
https://github.com/arthru/python-mk-livestatus
CentOS & Ubuntu
```
# pip install python-mk-livestatus
```

### CODE REVIEW

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+                "severity": hapi_severity,
+                "lastChangeTime": hapi_time,
+                "hostId": service["host_name"],
+                "hostName": service["host_alias"],
+                "brief": service["plugin_output"],
+                "extendedInfo": ""
+            })
+        self.__trigger_last_info = \
+            Utils.get_biggest_num_of_dict_array(triggers,
+                                                       "lastChangeTime")
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: Curious indentation.

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        result = query.call()
+
+        triggers = []
+        for service in result:
+            hapi_status, hapi_severity = \
+              self.__parse_status_and_severity(service["state"])
+
+            last_state_change = datetime.datetime.fromtimestamp(service["last_state_change"])
+            hapi_time = Utils.conv_to_hapi_time(last_state_change,
+                                                       self.__time_offset)
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: Curious indentation, too.

Comment (Mnakagawa) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: Oh... there were unexpected. I will fix its.

**Comment by cosmo0920** [comment]
pep8 complains following style issues:

```log
server/hap2/hatohol/hap2_nagios_livestatus.py:71:27: E701 multiple statements on one line (colon)
server/hap2/hatohol/hap2_nagios_livestatus.py:93:80: E501 line too long (88 > 79 characters)
server/hap2/hatohol/hap2_nagios_livestatus.py:100:11: E127 continuation line over-indented for visual indent
server/hap2/hatohol/hap2_nagios_livestatus.py:100:80: E501 line too long (87 > 79 characters)
server/hap2/hatohol/hap2_nagios_livestatus.py:106:80: E501 line too long (94 > 79 characters)
server/hap2/hatohol/hap2_nagios_livestatus.py:121:35: E701 multiple statements on one line (colon)
server/hap2/hatohol/hap2_nagios_livestatus.py:122:21: E701 multiple statements on one line (colon)
server/hap2/hatohol/hap2_nagios_livestatus.py:133:80: E501 line too long (100 > 79 characters)
server/hap2/hatohol/hap2_nagios_livestatus.py:134:80: E501 line too long (80 > 79 characters)
server/hap2/hatohol/hap2_nagios_livestatus.py:142:15: E121 continuation line indentation is not a multiple of four
server/hap2/hatohol/hap2_nagios_livestatus.py:144:80: E501 line too long (93 > 79 characters)
server/hap2/hatohol/hap2_nagios_livestatus.py:201:51: E701 multiple statements on one line (colon)
server/hap2/hatohol/hap2_nagios_livestatus.py:209:15: E127 continuation line over-indented for visual indent
```

Could you fix them?

File: server/hap2/hatohol/hap2_nagios_livestatus.py (hunk 2 above)
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: BTW, `it`'s plural form is `them`. :stuck_out_tongue_winking_eye:

File: hatohol.spec.in [hunk 3]
```diff
@@ -149,6 +149,14 @@ Requires: hatohol-hap2-common = %{version}
 %description hap2-nagios-ndoutils
 Nagios ndoutils plugin for hatohol arm plugins version 2.0.
 
+%package hap2-nagios-livestatus
+Summary: Nagios livestatus plugins of HAPI2.0
+Group: Applications/System
+Requires: hatohol-hap2-common = %{version}
```
Comment (cosmo0920) [inline hatohol.spec.in]: Should we add `mk-livestatus` and its python module in Requires?

**Comment by Mnakagawa** [comment]
We should not fix these breach of pep8 in this PR.
That caused from long haplib's function name and multiple statement.
These used in other hap2 plugin and library.
So if we fix it, we should create another PR.

**Comment by cosmo0920** [comment]
> We should not fix these breach of pep8 in this PR.

I don't think so.

> That caused from long haplib's function name and multiple statement.

pep8 warnings which is `E501 line too long` should not fix in this PR, but other warnings should be fix this PR or rebased PR.

File: hatohol.spec.in (hunk 3 above)
Comment (Mnakagawa) [inline hatohol.spec.in]: hap2-plugin can work on host that does not exist Nagios and check-mk-livestatus module.
So, we do not need to add check-mk-livestatus.
But python-mk-livestatus module is required. I install it in %post section.

File: server/hap2/hatohol/hap2_nagios_livestatus.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+from hatohol import hap
+from hatohol import haplib
+from hatohol.haplib import Utils
+from hatohol import standardhap
+import socket
+import uuid
+import json
+from mk_livestatus import Socket
+
+logger = getLogger(__name__)
```
Comment (cosmo0920) [inline server/hap2/hatohol/hap2_nagios_livestatus.py]: `__name__` should be replaced with `hatohol.nagios_livestatus` like string.

**Comment by Mnakagawa** [comment]
This PR is cousing conflict. So close and re-create.

**Pull request closed without merging**